site/static/home.js

```diff
@@ -27,8 +29,18 @@ function homePageSetup() {
 }
 
 function knockSocket(location) {
+  if (nameInput.value === "") {
+    displayNameless();
+    return;
+  }
+  console.log(location.protocol);
+  // If we're http, then we'll use an unencrypted websocket (for development)
+  let socketProtocol = 'wss';
+  if (location.protocol !== 'https:') {
```

Comment:
This will cause issues with the `location` parameter. Use `window.location.protocol` to refer to the global!